daemon/cmd/daemon_main.go

```diff
@@ -503,10 +503,6 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 	flags.Duration(option.KVstoreConnectivityTimeout, defaults.KVstoreConnectivityTimeout, "Time after which an incomplete kvstore operation  is considered failed")
 	option.BindEnv(vp, option.KVstoreConnectivityTimeout)
 
-	flags.Bool(option.KVstorePodNetworkSupport, defaults.KVstorePodNetworkSupport, "Enable the support for running the Cilium KVstore in pod network")
```

Comment:
Let's maybe add it to upgrade notes one more time to ensure that if someone was configuring this flag in v1.17 is aware that now it's noop?

Reply:
does passing a non existent flag act as a noop, or would that produce an error on its own?

Reply:
> Let's maybe add it to upgrade notes one more time to ensure that if someone was configuring this flag in v1.17 is aware that now it's noop?

I had thought about adding a release note, but ended up not doing that to avoid possible confusion. Indeed, the support removal has already been announced in v1.17, and this flag has always been hidden and undocumented. Happy to add a note if you feel that would be more appropriate though. 

> does passing a non existent flag act as a noop, or would that produce an error on its own?

It depends. It triggers a hard error if passed as a command line argument due to the flag not existing anymore, while it is silently ignored if specified via an environment variable or config map entry.